aws/gd_test.go

```diff
@@ -0,0 +1,64 @@
+package aws
+
+import (
+	"testing"
+	"time"
+
+	awsgo "github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/guardduty"
+	"github.com/gruntwork-io/go-commons/errors"
+	"github.com/stretchr/testify/assert"
+)
+
+// createTestGuardDuty creates a GuardDuty detector
+func createTestGuardDuty(t *testing.T, session *session.Session) string {
+	svc := guardduty.New(session)
+
+	detector, err := svc.CreateDetector(&guardduty.CreateDetectorInput{
+		Enable: awsgo.Bool(true),
+	})
+	if err != nil {
+		assert.Failf(t, "Could not create test GuardDuty instance", errors.WithStackTrace(err).Error())
+	}
+
+	return *detector.DetectorId
+}
+
+func TestNukeAllGuardDutyDetectors(t *testing.T) {
+	t.Parallel()
+
+	region, err := getRandomRegion()
+	if err != nil {
+		assert.Fail(t, errors.WithStackTrace(err).Error())
+	}
```

Comment:
```suggestion
	require.NoError(t, err)
```

Reply:
This is great, I haven't had much experience with the testify package so I was unaware of this! I will go through and change these.